angular-chess/src/app/components/chess-board.component.html

```diff
@@ -7,7 +7,7 @@
         [piece]="boardService.getPiece$(columnIndex, rowIndex) | async"
         [activePlayer]="boardService.activePlayer$ | async"
         [boardTheme]="themingService.selectedTheme$ | async"
-        [square]="highlightingService.getSquare$({ row: rowIndex, column: columnIndex })
+        [square]="highlightingService.getSquare$(SquareUtils.convertPositionToSquare({ row: rowIndex, column: columnIndex }))
```

Comment:
Looks like performance problem: Each field has a function call and a subscription.